crates/ruma-client-api/src/appservice/request_ping.rs

```diff
@@ -0,0 +1,64 @@
+//! `POST /_matrix/client/*/appservice/{appserviceId}/ping}`
+//!
+//! Ask the homeserver to ping the application service to ensure the connection works.
+
+pub mod v1 {
+    //! `/v1/` ([spec])
+    //!
+    //! [spec]: https://spec.matrix.org/latest/application-service-api/#post_matrixclientv1appserviceappserviceidping
+
+    use std::time::Duration;
+
+    use ruma_common::{
+        api::{request, response, Metadata},
+        metadata, OwnedTransactionId,
+    };
+
+    const METADATA: Metadata = metadata! {
+        method: POST,
+        rate_limited: false,
+        authentication: AccessToken,
+        history: {
+            unstable => "/_matrix/client/unstable/fi.mau.msc2659/appservice/:appservice_id/ping",
+            1.7 => "/_matrix/client/v1/appservice/:appservice_id/ping",
+        }
+    };
+
+    /// Request type for the `request_ping` endpoint.
+    #[request(error = crate::Error)]
+    pub struct Request {
+        /// The appservice ID of the appservice to ping.
+        ///
+        /// This must be the same as the appservice whose `as_token` is being used to authenticate
+        /// the request.
+        #[ruma_api(path)]
+        pub appservice_id: String,
+
+        /// An optional transaction ID that is passed through to the `POST /_matrix/app/v1/ping`
+        /// call.
```

Comment:
```suggestion
        /// Transaction ID that is passed through to the `POST /_matrix/app/v1/ping` call.
```

we don't need to state that it's optional, given it's an `Option`.